crates/telio-traversal/src/endpoint_providers/upnp/mod.rs

```diff
@@ -557,11 +582,13 @@ impl<Wg: WireGuard, I: UpnpEpCommands, E: Backoff> Runtime for State<Wg, I, E> {
                 let _ = self.handle_ping_rx(&buff[..len], &addr).await;
             }
             _ = &mut self.upnp_interval => {
-                if self.check_endpoint_candidate().await.is_ok(){
-                    self.upnp_interval =  PinnedSleep::new(self.exponential_backoff.get_backoff(), ());
-                } else {
-                    self.exponential_backoff.next_backoff();
-                };
+                if !self.is_endpoint_provider_paused {
```

Comment:
Can instead of polling we do similar thing like STUN Endpoint provider has, where it simply sleeps indefinately in a `tokio_select()`. It has `update` and `pending()`  arms where one is sleeping indefinately, and other processes `task_exec!`'s. Because pause/unpause is `task_exec()`. This would allow us to avoid unnecessary polling.